rundeckapp/grails-app/services/rundeck/services/ConfigurationService.groovy

```diff
@@ -212,4 +241,17 @@ class ConfigurationService {
             return defval
         }
     }
+
+    @Override
+    void afterPropertiesSet() throws Exception {
+        appCfg = grailsApplication?.config?.rundeck
+    }
+
+    Map<String,String> deprecatedKeyTrx = [
```

Comment:
maybe this map could be defined put into resources.groovy, or a config of its own, making it easier to extend later if more things come up, unless these are the only values we might need to translate

Reply:
As long as future configs are added to the typed configuration it should prevent devs from using config names that can't be mapped to Java property names.

Since these keys already exist, there needed to be some way of falling back from the new name to get the value out of the old if it exists. 

I hope this map doesn't need any additions.

Reply:
I would keep this mapping close to where properties are defined, to make easier for developers to find it. Probably this class is fine, or RundeckConfigBase. Also maybe changing this to an enum class could allow you to both define a constant, and map both old an new property keys to keep track of this legacy properties.